test-integration/operator-tests/src/test/java/org/infinispan/operator/ClientValidationIT.java

```diff
@@ -0,0 +1,101 @@
+package org.infinispan.operator;
+
+import cz.xtf.builder.builders.SecretBuilder;
+import cz.xtf.client.Http;
+import cz.xtf.core.http.Https;
+import cz.xtf.core.openshift.OpenShift;
+import cz.xtf.core.openshift.OpenShifts;
+import cz.xtf.junit5.annotations.CleanBeforeAll;
+import io.fabric8.kubernetes.api.model.Secret;
+import lombok.extern.slf4j.Slf4j;
+import org.assertj.core.api.Assertions;
+import org.infinispan.Infinispan;
+import org.infinispan.Infinispans;
+import org.infinispan.TestServer;
+import org.infinispan.identities.Credentials;
+import org.infinispan.util.CleanUpValidator;
+import org.infinispan.util.KeystoreGenerator;
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+
+import java.io.IOException;
+import java.net.URLEncoder;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Paths;
+
+/**
+ * ClientValidationIT tests if clients are validated. To validate client certificates,
+ * Data Grid requires a trust store that contains any part of the certificate chain for the signing authority,
+ * typically the root CA certificate.
+ * Any client that presents a certificate signed by the CA can connect to Data Grid.
+ */
+@Slf4j
+@CleanBeforeAll
+class ClientValidationIT {
+   private static final OpenShift openShift = OpenShifts.master();
+   private static final Infinispan infinispan = Infinispans.clientTlsValidation();
+   private static final TestServer testServer = TestServer.get();
+   private static final String testServerHost = testServer.host();
+
+   private static KeystoreGenerator.CertPaths certs;
+   private static String appName;
+   private static String hostName;
+   private static String user;
+   private static String pass;
+
+   @BeforeAll
+   static void deploy() throws IOException {
+      appName = infinispan.getClusterName();
+      hostName = openShift.generateHostname(appName + "-external");
+
+      certs = KeystoreGenerator.generateCerts(hostName, new String[]{appName});
+
+      Secret encryptionSecret = new SecretBuilder("encryption-secret")
+              .addData("keystore.p12", certs.keystore)
+              .addData("alias", hostName.getBytes())
+              .addData("password", "password".getBytes())
+              .build();
+
+      Secret authSecret = new SecretBuilder("connect-secret")
+              .addData("identities.yaml", Paths.get("src/test/resources/secrets/identities.yaml"))
+              .build();
+
+      openShift.secrets().create(encryptionSecret);
+      openShift.secrets().create(authSecret);
+
+      infinispan.deploy();
+      testServer.withSecret(appName + "-cert-secret").deploy();
```

Comment:
You are trying to mount secret that isn't expected to exist. You set in the configuration that the secret should be named `tls-secret`